### PR TITLE
Implement queue class, methods and tests

### PR DESCRIPTION
**Pull request by nodatall**
> The only journey is the one in the queue. 
~ Rainer Maria Rilke

### CODE REVIEW

File: spec/queue.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import Queue from '../src/queue'
+
+chai.use(chaiChange)
+
+describe('Queue', () => {
```
Comment (deonna) [inline spec/queue.js:7]: `length()` testing may be implied because you're using the `length()` method in some of your other functions, but it's a good idea to increase test coverage by adding a couple of standalone unit tests for `length()`: length of an empty queue and length of a populated queue.

File: src/queue.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+'use strict'
+
+export default class Queue {
+  constructor() {
+    this.storage = [],
+    this.head = 0,
```
Comment (deonna) [inline src/queue.js:6]: Nitpick: queue naming convention would be `this.front` and `this.back` instead of `this.head` and `this.end`.

Changing the way your `Queue` internals are named will also add consistency to your code.

File: src/queue.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    this.end = 0
+  }
+
+  enQueue( item ) {
```
Comment (deonna) [inline src/queue.js]: Another nitpick: en**q**ueue().

File: src/queue.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    this.storage[ this.end ] = item
+    this.end++
+  }
+
+  deQueue() {
```
Comment (deonna) [inline src/queue.js]: Another nitpick: de**q**ueue().

**Review by deonna — APPROVED** [review]
Good job - I <3 your naming.
